src/main/kotlin/org/openredstone/chad/commands/fractals.kt

```diff
@@ -0,0 +1,128 @@
+
+    val angle = rng.nextDouble(-3.14, 3.14)
+    val seedCoordinate = findGoodJulia(angle, messiness)
+
+    val a = rng.nextDouble(0.0, 0.2)
+    val b = rng.nextDouble(0.0, 0.2)
+    val c = rng.nextDouble(0.0, 0.2)
+
+    val img = BufferedImage(width, height, BufferedImage.TYPE_3BYTE_BGR)
+    for (y in 0 until height) {
+        for (x in 0 until ceil(width.toDouble() / 2.0).toInt()) {
+            val coX = aspectRatio * zoom * (x.toDouble() / width.toDouble() - 0.5)
+            val coY = zoom * (y.toDouble() / height.toDouble() - 0.5)
+            val coordinate = Complex(coX, coY)
+            val iterations = juliaPixel(coordinate, maxIterations, seedCoordinate)
+            val color = getColor(iterations, a, b, c)
+            img.setRGB(x, y, color)
+            img.setRGB(width - x - 1, height - y - 1, color)
+        }
+    }
+    return img
+}
+
+// Helper functions
+fun getColor(i: Double, a: Double, b: Double, c: Double): Int {
+    val red = max(sin(i * a) * 255.0, 0.0).toInt()
+    val green = max(sin(i * b) * 255.0, 0.0).toInt()
+    val blue = max(sin(i * c) * 255.0, 0.0).toInt()
+    return red + green.shl(8) + blue.shl(16)
+}
+
+fun juliaPixel(coordinate: Complex, maxIterations: Int, c: Complex): Double {
+    var z = coordinate
+    var i = 0
+    while (i < maxIterations && z.mag2() < 4.0) {
+        z = z.square() + c
+        i += 1
+    }
+    if (i < maxIterations) {
+        for (whyCantINameItUnderscore in 0 until 3) {
+            z = z.square() + c
+            i += 1
+        }
+        // actual magic
+        return i.toDouble() + 1.0 - ln(ln(sqrt(z.mag2()))) * 1.44269504089
+    }
+    return i.toDouble()
+}
+
+// room for improvement but works 99.9% of times (never crashes just gives up and returns a bad one)
+fun findGoodJulia(angle: Double, messiness: Int): Complex {
+    val x = cos(angle) * 2.0
+    val y = sin(angle) * 2.0
+    var coord = Complex(x, y)
+    var step = coord * 0.5
+    var sign = -1.0
+    var count = 0
+    while (count < 10_000) {
+        count += 1
+        coord += step * sign
+        val i = mandelPixel(coord, messiness + 1)
+        sign = if (i < messiness) {
+            -1.0
+        } else if (i > messiness) {
+            1.0
+        } else {
+            return coord
+        }
+        step *= 0.51
+    }
+    // bad luck, will get an almost blank image
+    return Complex(16.0, 0.0)
+}
+
+fun mandelPixel(coord: Complex, max_iterations: Int): Int {
+    var z = Complex(0.0, 0.0)
+    var i = 0
+    while (i < max_iterations && z.mag2() <= 4.0) {
+        z = z.square() + coord
+        i += 1
+    }
+    return i
+}
+
+// Hashing
+fun String.sha256lowerLong(): Long {
+    return MessageDigest
+        .getInstance("SHA-256")
+        .digest(this.toByteArray())
+        .fold(0) { acc, byte -> acc.shl(8).or(byte.toLong().and(255)) }
+}
+
+// Complex numbers
+data class Complex(val re: Double, val im: Double)
+
+operator fun Complex.plus(c: Complex): Complex {
```

Comment:
Same here with the short form, and you can also omit `this` to make it more concise (one liner)

Reply:
By the way, IntelliJ IDEA can automatically convert between the long (block body) and short (expression body) forms. You can bring up the menu with Alt+Enter (by default) when the cursor is in the function (or at least in the `return` keyword or the `=` character). That's very useful especially if you manage to simplify a multiline function into a single return statement (to convert to expression body) or if you need to add prints for debugging into a expression body function (convert to block body first, then add print)